File: hw/morillo/u3/Hw29 Composite, Observer, Strategy & Template/Composite/src/main/java/ec/edu/espe/composite/model/Employee.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.composite.model;

/**
 *
 * @author David Morillo, JavaSquad, DCCO-ESPE
 */
public abstract class Employee {
    String name = "not assigned yet";
    String title = "not assigned yet";
    
    public void stateName(){
        System.out.println(title + " " + name);
    }
}

File: hw/Rodriguez/u3/hw29/hw29Python/Composite/controller/Employee.py
from abc import ABC, abstractmethod

class Employee(ABC):
    def __init__(self, name="not assigned yet", title="not assigned yet"):
        self.name = name
        self.title = title

    def state_name(self):
        print(f"{self.title} {self.name}")

    @abstractmethod
    def add(self, employee):
        pass

    @abstractmethod
    def remove(self, employee):
        pass